File: src/knotlink-sdk-js/knotlink/SignalSender.js
/*
 * KnotLink SDK - JavaScript
 * Copyright (c) 2024-2026 KnotLink Contributors
 * SPDX-License-Identifier: MIT
 */

// SignalSender.js
const TcpClient = require('./TcpClient'); // 确保 TcpClient.js 在同一目录下

class SignalSender {
    constructor(APPID = null, SignalID = null) {
        this.appID = APPID;
        this.signalID = SignalID;
        this.KLsender = new TcpClient();
        this.KLsender.connectToServer("127.0.0.1", 6370,() => {
            console.log("OKK");
        });

    }

    setConfig(APPID, SignalID) {
        this.appID = APPID;
        this.signalID = SignalID;
    }

    emitt(data) {
        this._emitt(this.appID, this.signalID, data);
    }

    _emitt(APPID, SignalID, data) {
        if (typeof data === 'string') {
            data = Buffer.from(data, 'utf-8');
        }
        this.__emitt(APPID, SignalID, data);
    }

    __emitt(APPID, SignalID, data) {
        const sKey = `${APPID}-${SignalID}&*&`;
        const sKeyBytes = Buffer.from(sKey, 'utf-8');
        const sData = Buffer.concat([sKeyBytes, data]);
        this.KLsender.send_data(sData);
    }
}

// 导出 SignalSender 类
module.exports = SignalSender;

File: src/knotlink-sdk-python/knotlink/knotlink/SignalSender.py
# SignalSender.py
# KnotLink SDK - Python
# Copyright (c) 2024-2026 KnotLink Contributors
# SPDX-License-Identifier: MIT

from .tcpclient import TcpClient  # 确保 tcpclient.py 在同一目录下
import time

class SignalSender:
    def __init__(self, APPID=None, SignalID=None):
        self.appID = APPID
        self.signalID = SignalID
        self.KLsender = TcpClient()
        self.KLsender.connect_to_server("127.0.0.1", 6370)

    def set_config(self, APPID, SignalID):
        self.appID = APPID
        self.signalID = SignalID

    def emitt(self, data):
        self._emitt(self.appID, self.signalID, data)

    def _emitt(self, APPID, SignalID, data):
        if isinstance(data, str):
            data = data.encode('utf-8')
        self.__emitt(APPID, SignalID, data)

    def __emitt(self, APPID, SignalID, data):
        s_key = f"{APPID}-{SignalID}&*&"
        s_key_bytes = s_key.encode('utf-8')
        s_data = s_key_bytes + data
        self.KLsender.send_data(s_data)

# 示例用法
if __name__ == "__main__":
    signal_sender = SignalSender(APPID="1", SignalID="1")
    signal_sender.emitt("Hello, Server!")
    # 等待一段时间以确保数据发送完成
    time.sleep(10)
    signal_sender.KLsender.disconnect()